### DISCUSSION TITLE
Suggesting the track with notes playing if curent track has no notes on this measure(s)

### DISCUSSION THREAD

**Discussion by marriusco** [CoderLine/alphaTab#716] [General]
Is there a way to show the track, or find the track id that has notes currently playing
if the current rendered/selected track has no notes on few measures. 
Thank you

**Comment by Danielku15** [comment]
https://www.alphatab.net/docs/reference/api/playedbeatchanged/ is an event which is fired whenever internally it is detected that a new beat is played. is this sufficient for your needs?  a more fine grain detection might be available internally but its not exposed. 

whats the exact thing you plan do build?

**Reply by circinusX1** [reply]
_Hi, same me but my personal account. I was using my job account I did not noticed._
Thank you. I am using playedBeatChange to update the fret board already, but I was thinking
if current selected track has no notes on current measure(s) and other have, and you hear them,
to update the tracks pane with an indicator showing which track is actually have notes on the beat.
 Like this selection playing track 2:
[selection]( https://ghitara.com/index.php?play2=L/LessonsScales/LessonsScales-EasyArpeggiosTheory.gp4&genere=hard1.jpg&sse=492-713-1&ap=1&panel=topriff
)

**Reply by Danielku15** [reply]
I see... alphaTab only has the `playedBeatChange` which reports the played beat for the currently rendered tracks but not all of them. 

I dived again into the internals of the beat tracking I have made and I found out that alphaTab already [keeps track ](https://github.com/CoderLine/alphaTab/blob/190d4c54debe5cf95db5a00055a97825523f90cf/src/midi/BeatTickLookup.ts#L44)of the actively played beats independent from the renderered tracks and also across all voices and staves. 

It is used when [highlighting the beats](https://github.com/CoderLine/alphaTab/blob/190d4c54debe5cf95db5a00055a97825523f90cf/src/AlphaTabApiBase.ts#L966) which are currently played but it's not exposed as event. So it would only be a matter of adding an event like `activeBeatsChanged` which is called along the `playedBeatChanged` 

I added https://github.com/CoderLine/alphaTab/issues/722 as a concrete work item to add this functionality. My current focus for alphaTab is primarily to add native android and iOS support for 1.3 but I'm happy to accept contributions.

**Reply by circinusX1** [reply]
Hi Daniel, I Appreciate your answer. It helps a lot. Thank you very  much. Regards, Marius
I would be happy to contribute but ts script and java is not my strongest.
Anyway your suggestions helped a lot. My changes are marked mco-beat in the 
[alphaTab-1.3.0.js](https://ghitara.com/alpha_new/alphaTab-1.3.0.js).
The tracks with notes are blinking just fine for now indicating the sound. 
The track pane has to be enabled form leftmost bottom bar --- button.
[nonotes-on-track3_demo](https://ghitara.com/index.php?play2=I/InFlames/InFlames-OnlyForTheWeak.gp3&genere=unknown.jpg&sse=47-300-2&plpa&pmixer)   ... link fixed
